jetty-cdi/src/main/java/org/eclipse/jetty/cdi/CdiServletContainerInitializer.java

```diff
@@ -0,0 +1,59 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.cdi;
+
+import java.util.Set;
+import javax.servlet.ServletContainerInitializer;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+
+import org.eclipse.jetty.annotations.AnnotationConfiguration;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.webapp.WebAppContext;
+
+/**
+ * A {@link ServletContainerInitializer} that introspects for a CDI API
+ * implementation within a web application. If the CDI API is found, then
+ * a {@link CdiDecorator} is registered as a {@link org.eclipse.jetty.util.Decorator}
+ * for the context.
+ * @see AnnotationConfiguration.ServletContainerInitializerOrdering
+ */
+public class CdiServletContainerInitializer implements ServletContainerInitializer
+{
+    public static final String CDI_INTEGRATION_ATTRIBUTE = "org.eclipse.jetty.cdi";
+    private static final Logger LOG = Log.getLogger(CdiServletContainerInitializer.class);
+
+    @Override
+    public void onStartup(Set<Class<?>> c, ServletContext ctx) throws ServletException
+    {
+        try
+        {
+            WebAppContext context = WebAppContext.getCurrentWebAppContext();
+            context.getObjectFactory().addDecorator(new CdiDecorator(context));
```

Comment:
The `context` variable may be null so at this line we will bomb with a NPE.
Is it possible that `context` is null?
Is it possible that a `ServletContextHandler` is used instead and that this initializer is somehow run (perhaps manually)?

Reply:
SCI's really are mostly only meaningful with contexts.  I know we have repurposed some, but this one only makes sense with a context